tools/cmake/helpers/platform.cmake

```diff
@@ -74,20 +74,16 @@ function(get_os OS)
 endfunction()
 
 function(get_arch ARCH)
-  if(ARCH_X86)
+  if( CMAKE_SIZEOF_VOID_P MATCHES 8 )
+    # void ptr = 8 byte --> x86_64
+    set(${ARCH} "x86_64" PARENT_SCOPE) 
+  elseif( CMAKE_SIZEOF_VOID_P MATCHES 4 )
+    # void ptr = 4 byte --> x86
     set(${ARCH} "x86" PARENT_SCOPE)
-  elseif(ARCH_X64)
-    set(${ARCH} "x64" PARENT_SCOPE)
```

Comment:
Do you know whether the change from `x64` to `x86_64` will break anything? From what I've seen of the CMake changes, it looks like this `ARCH` value is only really relevant when building for Windows.